### ISSUE TITLE
Issue with Colorlayers

### ISSUE THREAD

**Issue by manuamador** [iTowns/itowns#1847]
Hi,
I have troubles with colorlayers from geojson, there is a rectangular patch that is changing with zooming but it doesn't move,
when zooming:
![image](https://user-images.githubusercontent.com/1217972/163547480-eff9277b-5fde-45f1-b59f-4448e2c699d5.png)
![image](https://user-images.githubusercontent.com/1217972/163547538-21ad238c-3660-436a-9a00-418f2ffe5238.png)
![image](https://user-images.githubusercontent.com/1217972/163547597-a5d3244f-b41d-4daa-b9c3-21578dd38cbf.png)

the code used to display the colorlayer is:
`const parcelles_dom = new itowns.FileSource({
                url: 'parcelles.geojson',
                crs: 'EPSG:4326',
                format: 'application/json',
            });
		
			
			function fillcolorDom(properties) {
                if (properties.PCM === 'P') {
                    return '#008000';
                } else if (properties.PCM === 'C') {
                    return '#ffa07a';
                } else if (properties.PCM === 'M') {
                    return '#da70d6';
                } else if (properties.PCM === '2') {
                    return '#666';
				} else if (properties.PCM === 'PF') {
                    return '#008000';
                } else if (properties.PCM === 'CF') {
                    return '#ffa07a';
                } else if (properties.PCM === 'MF') {
                    return '#da70d6';
                } else if (properties.PCM === '2F') {
                    return '#666';
                }
				else{
                return 'transparent';}
            }
			
			function strokecolorDom(properties) {
                if (properties.PCM === 'P') {
                    return '#008000';
                } else if (properties.PCM === 'C') {
                    return '#ffa07a';
                } else if (properties.PCM === 'M') {
                    return '#da70d6';
                } else if (properties.PCM === '2') {
                    return '#666';
				} else if (properties.PCM === 'PF') {
                    return '#008000';
                } else if (properties.PCM === 'CF') {
                    return '#ffa07a';
                } else if (properties.PCM === 'MF') {
                    return '#da70d6';
                } else if (properties.PCM === '2F') {
                    return 'white';
                }
				else {
                return 'white';}
            }
			
			
			
            const domlayer = new itowns.ColorLayer('Domanialité', {
                name: 'Domanialité',
                transparent: true,
                source: parcelles_dom,
                style: new itowns.Style({
                    fill: {
                        color: fillcolorDom,
                        opacity: 0.5,
                    },
                    stroke: {
                        color: strokecolorDom,
						width : 0.5,
                    },
                }),
            });
				view.addLayer(domlayer).then(function _() {
                    menuGlobe.addLayerGUI.bind(menuGlobe);
                    itowns.ColorLayersOrdering.moveLayerToIndex(view, 'Domanialité', 10);
                });`

Back to a previous version (2.32.x maybe) we had no problem.
I can PM a zip with the whole stuff for debugging.

**Comment by manuamador** [comment]
Hi, me again, I am stuck with this issue and it kills me to use an old version of the package (wiyhout the navigation button the search bar) to have a working view.
Can we expect a solution ?
As already expalined, I can provide anything to help you reproduce the problem.

**Comment by mgermerie** [comment]
Hi ! Sorry for the late response.
Could you please PM me a dataset so I can test it ? My email is `madec.germerie-guizouarn@ign.fr`. I will try to figure out where the issue comes from and will be able to tell you how and when we will be able to fix it.
Thanks in advance.

**Comment by mgermerie** [comment]
After digging, I found out that the bug appeared in [this commit](https://github.com/iTowns/itowns/pull/1609/commits/6ccad03b4041db9ab7815ef10b69024ed8a60dac).
The bug is the following : 

- if trying to display vector data in a `ColorLayer`, 
- if these vector data include points
- if no color is defined for points in the layer's `Style`

then the rendering of vector data for each tile that includes a point will fail.

In other words, it is mandatory to define a point style if we want to display some vector data that contains points.
I will make a PR to fix this. One should be able to chose not to display points if he/she does not define a style for them.

**Comment by manuamador** [comment]
Thanks! It works !
